bazel/BUILD.bazel

```diff
@@ -79,6 +79,17 @@ ic_version_or_git_sha(
     visibility = ["//visibility:public"],
 )
 
+genrule(
+    name = "commit_timestamp_txt",
+    outs = ["commit_timestamp.txt"],
+    cmd = "git show -s --format=%ct > $@",
+    tags = [
+        "local",
+        "no-cache",
+    ],
+    visibility = ["//visibility:public"],
+)
+
```

Comment:
Won't this cause the setupos to be rebuild for every commit?

I think it would be better to utilise https://bazel.build/docs/user-manual#workspace-status as in:
* return the commit time in: [bazel/workspace_status.sh](https://github.com/dfinity/ic/blob/master/bazel/workspace_status.sh) like
```
echo "COMMIT_TIME $(git show -s --format=%ct)"
```
* read the `ctx.version_file` when building the setupos 
* and extract `COMMIT_TIME`.